tests/src/JIT/Intrinsics/BinaryPrimitivesReverseEndianness.cs

```diff
@@ -0,0 +1,136 @@
+                Console.WriteLine($"Input:    0x{ConstantUInt32Input:X8}");
+                Console.WriteLine($"Output:   0x{swappedUInt32:X8}");
+                Console.WriteLine($"Expected: 0x{ConstantUInt32Expected:X8}");
+            }
+
+            ulong swappedUInt64 = BinaryPrimitives.ReverseEndianness(ConstantUInt64Input);
+            if (swappedUInt64 != ConstantUInt64Expected)
+            {
+                Console.WriteLine($"BinaryPrimitives.ReverseEndianness(const UInt32) failed.");
+                Console.WriteLine($"Input:    0x{ConstantUInt64Input:X16}");
+                Console.WriteLine($"Output:   0x{swappedUInt64:X16}");
+                Console.WriteLine($"Expected: 0x{ConstantUInt64Expected:X16}");
+            }
+
+            /*
+             * NON-CONST VALUE TESTS
+             */
+
+            ushort nonConstUInt16Input = (ushort)DateTime.UtcNow.Ticks;
+            ushort nonConstUInt16Output = BinaryPrimitives.ReverseEndianness(nonConstUInt16Input);
+            ushort nonConstUInt16Expected = ByteSwapUInt16_Control(nonConstUInt16Input);
+            if (nonConstUInt16Output != nonConstUInt16Expected)
+            {
+                Console.WriteLine($"BinaryPrimitives.ReverseEndianness(non-const UInt16) failed.");
+                Console.WriteLine($"Input:    0x{nonConstUInt16Input:X4}");
+                Console.WriteLine($"Output:   0x{nonConstUInt16Output:X4}");
+                Console.WriteLine($"Expected: 0x{nonConstUInt16Expected:X4}");
+            }
+
+            uint nonConstUInt32Input = (uint)DateTime.UtcNow.Ticks;
+            uint nonConstUInt32Output = BinaryPrimitives.ReverseEndianness(nonConstUInt32Input);
+            uint nonConstUInt32Expected = ByteSwapUInt32_Control(nonConstUInt32Input);
+            if (nonConstUInt32Output != nonConstUInt32Expected)
+            {
+                Console.WriteLine($"BinaryPrimitives.ReverseEndianness(non-const UInt32) failed.");
+                Console.WriteLine($"Input:    0x{nonConstUInt32Input:X8}");
+                Console.WriteLine($"Output:   0x{nonConstUInt32Output:X8}");
+                Console.WriteLine($"Expected: 0x{nonConstUInt32Expected:X8}");
+            }
+
+            ulong nonConstUInt64Input = (ulong)DateTime.UtcNow.Ticks;
+            ulong nonConstUInt64Output = BinaryPrimitives.ReverseEndianness(nonConstUInt64Input);
+            ulong nonConstUInt64Expected = ByteSwapUInt64_Control(nonConstUInt64Input);
+            if (nonConstUInt64Output != nonConstUInt64Expected)
+            {
+                Console.WriteLine($"BinaryPrimitives.ReverseEndianness(non-const UInt64) failed.");
+                Console.WriteLine($"Input:    0x{nonConstUInt64Input:X16}");
+                Console.WriteLine($"Output:   0x{nonConstUInt64Output:X16}");
+                Console.WriteLine($"Expected: 0x{nonConstUInt64Expected:X16}");
+            }
+
+            return Pass;
```

Comment:
@GrabYourPitchforks  I might be wrong, but it looks like this test never fails, even if something goes wrong. The code prints to the output but anyway returns `Pass` and `Fail` is never used.

Besides that this is a great PR and a great example for introducing new Intrinsics!